OAuth2/Client/IClient.cs

```diff
@@ -44,7 +44,7 @@ public interface IClient
         /// Callback request payload (parameters).
         /// <example>Request.QueryString</example>
         /// </param>
-        UserInfo GetUserInfo(NameValueCollection parameters);
+        UserInfo GetUserInfo(NameValueCollection parameters, NameValueCollection queryParameters = null);
```

Comment:
Can't we use `parameters` for this? I'm not sure we need to introduce a new field as all the other implementations are passing state when needed already.

Reply:
I think is the only form to pass parameters to the QueryUserInfo() method...

Reply:
Can you please confirm this as configuration is passed through to each oauth implementation. I know extra state parameters can be set on the initial request.

Reply:
I think it's only for OAuth1 clients, Twitter specific, but maybe another OAuth1 client needs this.

Reply:
@titarenko do you have any more info on this?

Reply:
Can you covert this to be a IDictionary<string,string>. It would be nice to not use any super specialized collections.